bootstrap

```diff
@@ -406,7 +410,8 @@ swig_git_commit_hash=613ff08150e6264a5a4a964f2badeb6bba221398
 omega_common_commit_hash=f5921b74bc4bf899d9a043ee3ecdbcbfd4bacb7b
 letor_common_commit_hash=f5921b74bc4bf899d9a043ee3ecdbcbfd4bacb7b
 
-if [ ! -r .git ] ; then
+gitdir=`git rev-parse --git-dir`
+if [ ! -r $gitdir ] ; then
   echo "$0: '.git' not found - this script should be run from a git repo cloned"
```

Comment:
The error message should probably use `$gitdir` too.

Reply:
Oh, except that plays out unhelpfully if we're not in a checkout:

```
$ git rev-parse --git-dir
fatal: not a git repository (or any of the parent directories): .git
```

So the user would then see:

```
$ ./bootstrap
fatal: not a git repository (or any of the parent directories): .git
./bootstrap: '' not found - this script should be run from a git repo cloned
```

I think we should now to able to rely on `git rev-parse --git-dir` to take care of checking we're actually in a git repo and to report the issue if not, and so we just need to check `$gitdir` isn't empty, and we can drop the `'.git' not found -` part from this error message.  Does that seem good to you?

Reply:
Yep, that seems fine.  I'm happy to update this patch and resend it, though it will have to wait for it not to be nearly midnight for me :)

FYI: I also ran into an issue where as a submodule it's not finding the right path to download autoconf and other modules, so I'll do some more testing with that too - though this bit is still clearly better than it was before for the submodule case.

Reply:
>  I'm happy to update this patch and resend it, though it will have to wait for it not to be nearly midnight for me :)

I guess this dropped off your radar.

I think you're better placed to finish this off than me as you actually use this configuration, so if you're still happy to update it that would be great.